core/src/main/scala/cell/HandlerPool.scala

```diff
@@ -134,7 +171,11 @@ class HandlerPool(parallelism: Int = 8, unhandledExceptionHandler: Throwable =>
     val result = key.resolve(cells)
 
     for ((c, v) <- result) {
-      cells.foreach(c.removeNextCallbacks(_))
+      cells.filterNot(_ == c).foreach(cell => {
+        // Note that there is a better solution for this in https://github.com/phaller/reactive-async/pull/58
```

Comment:
Do you mean we should first merge PR #58?

Reply:
#58 is small and the conflict that will arise can be handled easily. So we can focus on the lazy-PR first.